sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-integration.service.ts

```diff
@@ -211,12 +215,92 @@ export class ECertFullTimeIntegrationService extends SFTPIntegrationBase<void> {
+   * E-Cert response folder.
+   */
+  async getResponseFilesFullPath(): Promise<string[]> {
+    let filesToProcess: Client.FileInfo[];
+    const client = await this.getClient();
+    try {
+      filesToProcess = await client.list(
+        `${this.esdcConfig.ftpResponseFolder}`,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.CERTSFB.*\\.DAT`,
+          "i",
+        ),
+      );
+    } finally {
+      await SshService.closeQuietly(client);
+    }
+    return filesToProcess.map(
+      (file) => `${this.esdcConfig.ftpResponseFolder}/${file.name}`,
+    );
+  }
+
+  // Generate Record
+
   /**
    * Transform the text lines in parsed objects specific to the integration process.
-   * TODO: read the feedback file.
    * @param remoteFilePath full remote file path with file name.
+   * @returns Parsed records from the file.
    */
-  downloadResponseFile(remoteFilePath: string): Promise<void> {
-    throw new Error("To be implemented.");
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<ECertResponseRecord[]> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    /**
+     * Read the first line to check if the header code is the expected one.
+     * and remove header.
+     */
+    const header = ECertFileHeader.createFromLine(fileLines.shift());
+    if (header.recordTypeCode !== RecordTypeCodes.ECertHeader) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has an invalid transaction code on header: ${header.recordTypeCode}`,
+      );
+      // If the header is not the expected one.
+      throw new Error(
+        "The E-Cert file has an invalid transaction code on header",
+      );
+    }
+
+    /**
+     * Read the last line to check if the trailer code is the expected one
+     * and remove trailer line.
+     */
+    const trailer = ECertFileFooter.createFromLine(fileLines.pop());
+    if (trailer.recordTypeCode !== RecordTypeCodes.ECertFooter) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has an invalid transaction code on trailer: ${trailer.recordTypeCode}`,
+      );
+      // If the trailer is not the expected one.
+      throw new Error(
+        "The E-Cert file has an invalid transaction code on trailer",
+      );
+    }
+
+    /**
+     * Check if the number of records match the trailer record count
+     * Here total record count is the total records rejected
+     */
+    if (trailer.recordCount !== fileLines.length) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has invalid number of records, expected ${trailer.recordCount} but got ${fileLines.length}`,
+      );
+      // If the number of records does not match the trailer record count..
+      throw new Error("The E-Cert file has invalid number of records");
+    }
+
+    // Generate the records.
+    const feedbackRecords: ECertResponseRecord[] = [];
+    fileLines.forEach((line: string, index: number) => {
```

Comment:
The result here could be simplified to:
```
return fileLines.map(
      (line: string, index: number) => new ECertResponseRecord(line, index + 1),
    );
```

Reply:
As we talked about, the code now has the SIN hash calculation, so please ignore this comment.